src/transformers/keras_callbacks.py

```diff
@@ -16,6 +18,132 @@
 logger = logging.getLogger(__name__)
 
 
+class KerasMetricCallback(Callback):
+    """
+    Callback to prompt metrics at the end of every epoch.
+
+    Args:
+        metric_fn: Metric function provided by the user.
+        val_dataset: Validation data to be used to evaluate the model at
+        the end of the epoch.
+        batch_size: Batch size.
+        labels: Labels.
+    """
+
+    def __init__(
+        self,
+        metric_fn: Callable,
+        val_dataset: Union[tf.data.Dataset, np.ndarray, tf.Tensor, tuple, dict],
```

Comment:
The trainer uses `eval_dataset` but maybe Keras uses `val_dataset` more consistently?

Reply:
I'm fine to change this to `eval_dataset` for internal consistency!